stan/math/prim/prob/multi_student_t_cholesky_lpdf.hpp

```diff
@@ -0,0 +1,149 @@
+#ifndef STAN_MATH_PRIM_PROB_MULTI_STUDENT_T_CHOLESKY_LPDF_HPP
+#define STAN_MATH_PRIM_PROB_MULTI_STUDENT_T_CHOLESKY_LPDF_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/as_column_vector_or_scalar.hpp>
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/fun/dot_self.hpp>
+#include <stan/math/prim/fun/is_inf.hpp>
+#include <stan/math/prim/fun/log.hpp>
+#include <stan/math/prim/fun/log1p.hpp>
+#include <stan/math/prim/fun/lgamma.hpp>
+#include <stan/math/prim/fun/max_size_mvt.hpp>
+#include <stan/math/prim/fun/size_mvt.hpp>
+#include <stan/math/prim/fun/to_ref.hpp>
+#include <stan/math/prim/fun/vector_seq_view.hpp>
+#include <stan/math/prim/prob/multi_normal_cholesky_lpdf.hpp>
+#include <cmath>
+#include <cstdlib>
+
+namespace stan {
+namespace math {
+
+/** \ingroup multivar_dists
+ * Return the log of the multivariate Student t distribution
+ * at the specified arguments.
+ *
+ * @tparam propto Carry out calculations up to a proportion
+ */
+template <bool propto, typename T_y, typename T_dof, typename T_loc,
+          typename T_scale>
+return_type_t<T_y, T_dof, T_loc, T_scale> multi_student_t_cholesky_lpdf(
+    const T_y& y, const T_dof& nu, const T_loc& mu, const T_scale& L) {
+  using T_scale_elem = typename scalar_type<T_scale>::type;
+  using lp_type = return_type_t<T_y, T_dof, T_loc, T_scale>;
+  using Eigen::Matrix;
+  using std::log;
+  using std::vector;
+  static const char* function = "multi_student_t_cholesky";
+  check_not_nan(function, "Degrees of freedom parameter", nu);
+  check_positive(function, "Degrees of freedom parameter", nu);
+
+  if (is_inf(nu)) {
```

Comment:
This is left over from the non-cholesky version. I think we should accept this current form because it follows the same standard as the non-cholesky version and then create a separate issue to remove this in both.

Reply:
I take that back, I'm removing. If I use `inv_gamma_rng` it does not accept an `inf` (however boost's gamma does). So this should be consistent with our internal `inv_gamma_rng` function.